### PR TITLE
2496 file encrypted event

### PR DESCRIPTION
**Pull request by ilija-lazoroski**
# What does this PR do?

Fixes part of #2496

## PR Checklist
* [x] Have you added an explanation of what your changes do and why you'd like to include them?
* [x] Is the TravisCI build passing?
* [x] Was the CHANGELOG.md updated to reflect the changes?
* [x] Was the documentation framework updated to reflect the changes?
* [x] Have you checked that you haven't introduced any duplicate code?

## Testing Checklist

* [x] Added relevant unit tests?
* [x] Have you successfully tested your changes locally? Elaborate:
    > Tested by UTs
* [x] If applicable, add screenshots or log transcripts of the feature working

### CODE REVIEW

File: monkey/common/agent_events/file_encrypted_event.py [hunk 1]
```diff
@@ -0,0 +1,23 @@
+from pathlib import PurePath
+
+from pydantic import Field
+
+from . import AbstractAgentEvent
+
+
+class FileEncryptedEvent(AbstractAgentEvent):
+    """
+    An event that occurs when the Agent encrypts a file.
```
Comment (shreyamalviya) [inline monkey/common/agent_events/file_encrypted_event.py:10]:
```suggestion
    An event that occurs when the Agent encrypts a file
```

**Review by shreyamalviya — COMMENTED** [review]
Is there any reason we're not adding validators like the pydantic issue talks about?

File: monkey/common/agent_event_serializers/register.py [hunk 2]
```diff
@@ -2,6 +2,7 @@
     AgentShutdownEvent,
     CredentialsStolenEvent,
     ExploitationEvent,
+    FileEncryptedEvent,
```
Comment (VakarisZ) [inline monkey/common/agent_event_serializers/register.py:5]: Looks like we already broke consistency. If there are `ExploitationEvent`, `PasswordResorationEvent`, then this should probably be `FileEncryptionEvent`

File: monkey/common/agent_events/file_encrypted_event.py [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+    """
+    An event that occurs when the Agent encrypts a file
+
+    Attributes:
+        :param file_path: Path of the encrypted file
+        :param success: Status of the file encryption
+        :param error_message: Message if an error occurs during file encryption
+    """
+
+    file_path: PurePath
```
Comment (VakarisZ) [inline monkey/common/agent_events/file_encrypted_event.py:34]: I would've just used a string. If the users of the event need `PurePath` so desperatelly, they can fetch the OS from the machine.

File: monkey/common/agent_events/file_encrypted_event.py [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+
+        if not isinstance(os_string, str):
+            raise TypeError("Operating system must be a string")
+
+        if os_string == OperatingSystem.LINUX.value:
+            return PurePosixPath(path_string)
+        elif os_string == OperatingSystem.WINDOWS.value:
+            return PureWindowsPath(path_string)
+
+        raise ValueError(f'"{os_string}" is not a valid operating system')
```
Comment (VakarisZ) [inline monkey/common/agent_events/file_encrypted_event.py:60]: We can avoid this boilerplate code by adding `source_os: OperatingSystem` to `FileEncryptedEvent` Then we can use [root validator](https://pydantic-docs.helpmanual.io/usage/validators/#root-validators) to cast it to whatever path we need based on OS

**Review by shreyamalviya — CHANGES_REQUESTED** [review]
Register the new event in `common/agent_events/register.py` too.

File: monkey/common/agent_events/file_encrypted_event.py (hunk 4 above)
Comment (mssalvatore) [reply monkey/common/agent_events/file_encrypted_event.py:60]: Discussed on slack:

The OS is metadata about the path, not data about the event. The cost is some extra code, the benefit is the event isn't polluted with metadata.

**Review by mssalvatore — APPROVED** [review]

**Pull request merged**